test/gtest/uct/test_peer_failure.cc

```diff
@@ -280,32 +307,41 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure, purge_failed_peer,
     send_recv_am(0);
     send_recv_am(1);
 
-    const ucs_time_t loop_end_limit = ucs::get_deadline();
     const size_t num_pend_sends     = 64ul;
-    const uint64_t send_data        = 0;
+    const ucs_time_t loop_end_limit = ucs::get_deadline();
     std::vector<pending_send_request_t> reqs(num_pend_sends);
 
     {
+        bool receiver_killed = false;
         scoped_log_handler slh(wrap_errors_logger);
-
         ucs_status_t status;
+
         do {
-            status = uct_ep_am_short(ep0(), 0, 0, &send_data,
-                                     sizeof(send_data));
-        } while ((status == UCS_OK) && (ucs_get_time() < loop_end_limit));
+            fill_resources(receiver_killed, loop_end_limit);
 
-        if (status == UCS_OK) {
-            UCS_TEST_SKIP_R("unable to fill the UCT resources");
-        } else if (status != UCS_ERR_NO_RESOURCE) {
-            UCS_TEST_ABORT("AM Short failed with " << ucs_status_string(status));
-        }
+            if (!receiver_killed) {
+                kill_receiver();
```

Comment:
IMHO uniting it to one loop is confusing, better have separate code to fill+kill before the add_pending+fill loop

Reply:
I see
done